{"name": "test_invalid_login_shows_error", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.InvalidSelectorException: Message: Locator Strategy '-ios predicate string' is not supported for this session; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#invalidselectorexception\nStacktrace:\nInvalidSelectorError: Locator Strategy '-ios predicate string' is not supported for this session\n    at AndroidUiautomator2Driver.validateLocatorStrategy (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:393:13)\n    at AndroidUiautomator2Driver.findElOrElsWithProcessing (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/commands/find.js:31:8)\n    at AndroidUiautomator2Driver.findElement (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/commands/find.js:53:21)\n    at commandExecutor (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:335:9)\n    at /nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:146:12\n    at AsyncLock._promiseTry (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:280:31)\n    at exec (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:145:9)\n    at AsyncLock.acquire (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:162:3)\n    at AndroidUiautomator2Driver.executeCommand (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:348:39)\n    at AppiumDriver.executeCommand (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/lib/appium.js:563:36)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n    at asyncHandler (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-base-driver/lib/protocol/protocol.js:297:21)", "trace": "self = <tests.test_login.TestLogin object at 0x00000240FC8EB250>\ndriver = <appium.webdriver.webdriver.WebDriver (session=\"68fbc02f85447f20a41548002f45bdd67c3b1f2a\")>\nwait = <selenium.webdriver.support.wait.WebDriverWait (session=\"68fbc02f85447f20a41548002f45bdd67c3b1f2a\")>\n\n    def test_invalid_login_shows_error(self, driver, wait):\n        lp = LoginPage(driver, wait)\n        lp.login(\"wrong_user\", \"wrong_pass\")\n    \n        # make sure we are NOT logged in\n>       assert not lp.is_logged_in(timeout=3), \"Unexpectedly logged in with invalid credentials\"\n\ntests\\test_login.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages\\login_page.py:62: in is_logged_in\n    wait.until(EC.presence_of_element_located(loc))\nC:\\Users\\db595\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:129: in until\n    value = method(self._driver)\nC:\\Users\\db595\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\support\\expected_conditions.py:104: in _predicate\n    return driver.find_element(*locator)\nC:\\Users\\db595\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:926: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\nC:\\Users\\db595\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:458: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <appium.webdriver.errorhandler.MobileErrorHandler object at 0x00000240FCBF3380>\nresponse = {'status': 400, 'value': '{\"value\":{\"error\":\"invalid selector\",\"message\":\"Locator Strategy \\'-ios predicate string\\' i...re/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-base-driver/lib/protocol/protocol.js:297:21)\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"\n        https://www.w3.org/TR/webdriver/#errors\n        \"\"\"\n        payload = response.get('value', '')\n        if isinstance(payload, dict):\n            payload_dict = payload\n        else:\n            try:\n                payload_dict = json.loads(payload)\n            except (json.JSONDecodeError, TypeError):\n                return\n            if not isinstance(payload_dict, dict):\n                return\n        value = payload_dict.get('value')\n        if not isinstance(value, dict):\n            return\n        error = value.get('error')\n        if not error:\n            return\n    \n        message = value.get('message', error)\n        stacktrace = value.get('stacktrace', '')\n        # In theory, we should also be checking HTTP status codes.\n        # Java client, for example, prints a warning if the actual `error`\n        # value does not match to the response's HTTP status code.\n        exception_class: Type[sel_exceptions.WebDriverException] = ERROR_TO_EXC_MAPPING.get(\n            error, sel_exceptions.WebDriverException\n        )\n        if exception_class is sel_exceptions.WebDriverException and message:\n            if message == 'No such context found.':\n                exception_class = appium_exceptions.NoSuchContextException\n            elif message == 'That command could not be executed in the current context.':\n                exception_class = appium_exceptions.InvalidSwitchToTargetException\n    \n        if exception_class is sel_exceptions.UnexpectedAlertPresentException:\n            raise sel_exceptions.UnexpectedAlertPresentException(\n                msg=message,\n                stacktrace=format_stacktrace(stacktrace),\n                alert_text=value.get('data'),\n            )\n>       raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))\nE       selenium.common.exceptions.InvalidSelectorException: Message: Locator Strategy '-ios predicate string' is not supported for this session; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#invalidselectorexception\nE       Stacktrace:\nE       InvalidSelectorError: Locator Strategy '-ios predicate string' is not supported for this session\nE           at AndroidUiautomator2Driver.validateLocatorStrategy (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:393:13)\nE           at AndroidUiautomator2Driver.findElOrElsWithProcessing (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/commands/find.js:31:8)\nE           at AndroidUiautomator2Driver.findElement (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/commands/find.js:53:21)\nE           at commandExecutor (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:335:9)\nE           at /nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:146:12\nE           at AsyncLock._promiseTry (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:280:31)\nE           at exec (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:145:9)\nE           at AsyncLock.acquire (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/async-lock/lib/index.js:162:3)\nE           at AndroidUiautomator2Driver.executeCommand (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-uiautomator2-driver/node_modules/appium-base-driver/lib/basedriver/driver.js:348:39)\nE           at AppiumDriver.executeCommand (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/lib/appium.js:563:36)\nE           at runMicrotasks (<anonymous>)\nE           at processTicksAndRejections (internal/process/task_queues.js:97:5)\nE           at asyncHandler (/nix/store/7gcl75xhklpjnw78s382pkyxgviyrsk4-appium-1.22.0/node_modules/appium-base-driver/lib/protocol/protocol.js:297:21)\n\nC:\\Users\\db595\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\appium\\webdriver\\errorhandler.py:125: InvalidSelectorException"}, "start": 1758106273475, "stop": 1758106288100, "uuid": "9216c029-1f09-4eb2-9bbd-f1ec05864281", "historyId": "e09b804d5fa8ae0d9600e7c83287794d", "testCaseId": "f37fcd220b67e01a4c24c167768af050", "fullName": "tests.test_login.TestLogin#test_invalid_login_shows_error", "labels": [{"name": "tag", "value": "smoke"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "DB-L-174"}, {"name": "thread", "value": "10772-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_login"}]}